westcoast/src/main/java/frc/commands/pidDriveAHRS.java

```diff
@@ -0,0 +1,72 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.commands;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class pidDriveAHRS extends CommandBase {
+  private DifferentialDrive drive;
+  private PIDController pid;
+  private Joystick joy;
+  private AHRS ahrs;
+
+  public pidDriveAHRS(
+    DifferentialDrive drive,
+    Joystick joystick,
+    AHRS ahrs
+  ) {
+    this.ahrs = ahrs;
+    this.drive = drive;
+    this.joy = joystick;
+  }
+
+  @Override
+  public void initialize() {
+    pid.reset();
+    pid.setTolerance(0.3);
+    pid.disableContinuousInput();
+
+    double P = SmartDashboard.getNumber("P", 0.1);
+    double I = SmartDashboard.getNumber("I", 0.1);
+    double D = SmartDashboard.getNumber("D", 0.1);
+    pid = new PIDController(P, I, D);
+  }
+
+  @Override
+  public void execute() {
+    double Rotate = pid.calculate(ahrs.getRate(), attenuate(joy.getTwist()));
+    drive.arcadeDrive(attenuate(joy.getY()), Rotate, false);
+
+    SmartDashboard.putBoolean("On Target?", pid.atSetpoint());
+    SmartDashboard.putNumber("Error", pid.getPositionError());
+    SmartDashboard.putNumber("Value", Rotate);
+    SmartDashboard.putNumber("P", pid.getP());
+    SmartDashboard.putNumber("I", pid.getI());
+    SmartDashboard.putNumber("D", pid.getD());
```

Comment:
This values are not changed by the system. getting this values and applying to the pid instance would be more useful otherwise they are only wasting network. for debug purposes it is fine.